pkgs/os-specific/linux/amdgpu-pro/default.nix

```diff
@@ -0,0 +1,146 @@
+  ];
+
+  patchPhase = optionalString (!libsOnly) ''
+    pushd usr/src/amdgpu-pro-${build}
+    for patch in $modulePatches; do echo $patch; patch -p1 < $patch; done
+    popd
+  '';
+
+  preBuild = optionalString (!libsOnly) ''
+    makeFlags="$makeFlags M=$(pwd)/usr/src/amdgpu-pro-${build}"
+  '';
+
+  postBuild = optionalString (!libsOnly) ''
+    xz usr/src/amdgpu-pro-${build}/amd/amdgpu/amdgpu.ko
+  '';
+
+  makeFlags = optionalString (!libsOnly)
+    "-C ${kernel.dev}/lib/modules/${kernel.modDirVersion}/build modules";
+
+  depLibPath = makeLibraryPath [
+    stdenv.cc.cc.lib xorg.libXext xorg.libX11 xorg.libXdamage xorg.libXfixes zlib
+    xorg.libXxf86vm libxcb libxshmfence epoxy openssl libdrm elfutils libudev
+  ];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r usr/bin $out/bin
+    cp -r etc $out/etc
+    cp -r usr/include $out/include
+    cp -r usr/lib/${libArch} $out/lib
+    mv $out/lib/amdgpu-pro/* $out/lib/
+    rmdir $out/lib/amdgpu-pro
+    cp -r usr/share $out/share
+  '' + optionalString (!libsOnly) ''
+    rm $out/lib/xorg
+    mv $out/lib/1.18 $out/lib/xorg
+    rm -r $out/lib/1.*
+    cp -r lib/firmware $out/lib/firmware
+    mkdir -p $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/gpu/drm/amd/amdgpu/amdgpu.ko.xz
+    cp usr/src/amdgpu-pro-${build}/amd/amdgpu/amdgpu.ko.xz $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/gpu/drm/amd/amdgpu/amdgpu.ko.xz
+  '' + ''
+    interpreter="$(cat $NIX_CC/nix-support/dynamic-linker)"
+    libPath="$out/lib:$out/lib/gbm:$depLibPath"
+    echo patching with $interpreter $libPath
+    for prog in "$out"/bin/*; do
+      echo patching program $prog
+      patchelf --interpreter "$interpreter" --set-rpath "$libPath" "$prog"
+    done
+    for lib in `find "$out/lib/" -name '*.so*'`; do
+      echo patching library $lib
+      patchelf --set-rpath "$libPath" "$lib"
+    done
+  '';
+
+  postFixup = ''
+    libPath="$out/lib:$out/lib/gbm:$depLibPath"
+    for lib in libgbm.so.1.0.0 ${optionalString (!libsOnly) "xorg/modules/drivers/amdgpu_drv.so"} amdvlk${bitness}.so vdpau/libvdpau_amdgpu.so; do
+        patchelf --set-rpath "$libPath" "$out/lib/$lib"
+    done
+    for lib in libEGL.so.1 libGL.so.1.2 ${optionalString (!libsOnly) "xorg/modules/extensions/libglx.so"} dri/amdgpu_dri.so; do
+      perl -pi -e 's:/usr/lib/${libArch}/dri:${driverLink}/lib/dri\0\0\0:g' "$out/lib/$lib"
+      perl -pi -e 's:/usr/lib/${libArch}:${driverLink}/lib\0\0\0:g' "$out/lib/$lib"
+    done
+    sed -i "s:/usr/lib/${libArch}:$out/lib:g" "$out/etc/vulkan/icd.d/amd_icd${bitness}.json"
+  '';
+
+  buildInputs = [
+    patchelf
+    perl
+  ];
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "AMDGPU-PRO drivers";
+    homepage =  http://support.amd.com/en-us/kb-articles/Pages/AMDGPU-PRO-Beta-Driver-for-Vulkan-Release-Notes.aspx ;
+    license = licenses.unfree;
+    platforms = platforms.linux;
+    hydraPlatforms = [];
```

Comment:
Better drop the `hydraPlatforms` line, unless there's some other reason than `license`. Hydra does license filtering by itself.